### PR TITLE
Tome rework (#537)

### PR DESCRIPTION
**Pull request by AustraliaIsMyCity**
Removed tome of knowledge from the shop
Tomes are now automatically given to the 5 lowest XP players from each team every 10 minutes
This excludes abandoned players

Fixes: #537

### CODE REVIEW

**Comment by crash0r** [comment]
Well, how are you changing game stuff when ONE player does request this? What about all the people, which are against?

**Comment by AustraliaIsMyCity** [comment]
> Well, how are you changing game stuff when ONE player does request this? What about all the people, which are against? That's a really strange style of working with the community...

we discussed it internally and all agreed its a good idea

**Comment by GeneralDotV2** [comment]
Almost like that "ONE player" actually knows what's best for 12v12 eyy.

**Comment by vjt4** [comment]
This change should be a vote at the loading screen

**Comment by AustraliaIsMyCity** [comment]
Nah

File: game/scripts/vscripts/addon_game_mode.lua [hunk 1]
```diff
@@ -1126,9 +1125,45 @@ function CMegaDotaGameMode:OnGameRulesStateChange(keys)
...
 	if newState == DOTA_GAMERULES_STATE_GAME_IN_PROGRESS then
-		
+
 		ShuffleTeam:GiveBonusToWeakTeam()
 
+		-- Add tome of knowledge to 5 lowest XP players on each team
+		Timers:CreateTimer(600, function()
+			local players = {[2] = {}, [3] = {}}
+
+			for player_id = 0, 23 do
```
Comment (Sheodar) [inline game/scripts/vscripts/addon_game_mode.lua:1135]: You no need find players in teams each iteration.
U can do it one time just out of timer

File: game/scripts/vscripts/addon_game_mode.lua [hunk 2]
```diff
@@ -1126,9 +1125,45 @@ function CMegaDotaGameMode:OnGameRulesStateChange(keys)
...
+
+			for team_id = 2, 3 do
+				-- Sort table from lowest xp to highest
+				table.sort(players[team_id], function(a, b)
+					return PlayerResource:GetTotalEarnedXP(a) < PlayerResource:GetTotalEarnedXP(b)
+				end)
+
+				local count = 5
+
+				for _, player_id in pairs(players[team_id]) do
```
Comment (Sheodar) [inline game/scripts/vscripts/addon_game_mode.lua:1151]: U can do same without outside counter.
Just like 
```
for counter = 1, 5 do
  if (players[team_id][counter]) then
    ...
  end
end
```

File: game/scripts/vscripts/addon_game_mode.lua (hunk 1 above)
Comment (AustraliaIsMyCity) [reply game/scripts/vscripts/addon_game_mode.lua:1135]: good point, will move it outside loop and also add a check to make sure player isn't abandoned to make sure we aren't giving tomes to abandoned players

File: game/scripts/vscripts/addon_game_mode.lua (hunk 2 above)
Comment (AustraliaIsMyCity) [reply game/scripts/vscripts/addon_game_mode.lua:1151]: smart, will do

Comment (AustraliaIsMyCity) [reply game/scripts/vscripts/addon_game_mode.lua:1151]: then again, itll be easier to implement excluding abandoned players this way so i will keep it

**Pull request merged**